design/API_requirements_and_design.md

```diff
@@ -0,0 +1,59 @@
+# Thin-Edge API requirements and design
+
+One should be able to build a thin-edge executable from tedge plugins crates that have been implemented independently:
+
+* All these tedge plugins crates depends on the tedge_api crate that defines how
+  to create, connect and run plugin instances.
+* Each tedge plugin crate can define its own types of messages. For instance, a
+  tedge_telemetry plugin can defines measurements, events, alarms and setpoints
+  messages.
+* To exchange typed messages, plugin crates need dependency relationship. For
+  instance, the tedge_json_over_mqtt plugin crate will have to depend on the
+  tedge_telemetry plugin crate and the tedge_mqtt crate.
+* The tedge_api crate defines only messages related to the runtime as Shutdown.
```

Comment:
```suggestion
* The tedge_api crate defines only messages related to the runtime as as `Shutdown`.
```

Reply:
Not sure to grasp the point.

Reply:
Hm, somehow this suggestion got messed up. Sorry.